clients/client/src/main/java/org/projectnessie/client/http/HttpRuntimeConfig.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.http;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.net.URI;
+import java.util.List;
+import java.util.Objects;
+import javax.net.ssl.SSLContext;
+
+/** Package-private HTTP configuration holder. */
+final class HttpRuntimeConfig {
```

Comment:
nit, maybe rename to `HttpClientConfig`?

Reply:
Explicitly named it like this as it's a mix of configuration from the builder(s) plus dynamic registrations (the request+response filters).